### ISSUE TITLE
run repo2docker in CI

### ISSUE THREAD

**Issue by lukasheinrich** [scikit-hep/pyhf#918]
# Description

right now we trigger a binder build but i'm not sure, do we get a response / feedback iif the bulid fails? We could use repo2docker to do the binder-like build locally

https://twitter.com/m_deff/status/1277064783662141441

**Comment by matthewfeickert** [comment]
> Right now we trigger a binder build but i'm not sure, do we get a response / feedback if the bulid fails?

No. `trigger_binder.sh` just `curls` the endpoint and gives the return code

https://github.com/scikit-hep/pyhf/blob/64dbce070156945f6bb524f79ee6e23d26b52f01/binder/trigger_binder.sh#L6-L16

and as we're using the `postBuild` Binder config which is basically just installing how we would in CI

https://github.com/scikit-hep/pyhf/blob/64dbce070156945f6bb524f79ee6e23d26b52f01/binder/postBuild#L1-L2

the `repo2docker` build on the Binder side shouldn't fail. That being said though, the idea of using `repo2docker` in CI (to test and prebuild the Docker image for Binder so that work was offloaded from them) were the reasons that I was looking at [`machine-learning-apps/repo2docker-action`](https://github.com/machine-learning-apps/repo2docker-action). The discussions with Tim and Chris on these Issues:

- [`machine-learning-apps/repo2docker-action` Issue 26](https://github.com/machine-learning-apps/repo2docker-action/issues/26)
- [`machine-learning-apps/repo2docker-action` Issue 25](https://github.com/machine-learning-apps/repo2docker-action/issues/25)

make it seem like it might not be a good fit for us as we want to use Binder config files in the top level `binder/` dir.

@lukasheinrich did you have additional ideas related to this?
